tests/testscript.py

```diff
@@ -1246,6 +1246,31 @@ def test_live_migration_with_vcpu_pinning(self):
         assert int(taskset_vcpu0_controller, 16) == int(taskset_vcpu0_compute, 16)
         assert int(taskset_vcpu2_controller, 16) == int(taskset_vcpu2_compute, 16)
 
+    def test_live_migration_tls(self):
+        """
+        Test the TLS encrypted live migration via virsh between 2 hosts. With
+        the right certificates, using IP addresses and DNS names should work,
+        thus we test all combinations.
+        """
+
+        controllerVM.succeed("virsh define /etc/domain-chv.xml")
+        controllerVM.succeed("virsh start testvm")
+
+        assert wait_for_ssh(controllerVM)
+
+        parallel_string = "--parallel --parallel-connections 4"
+        for parallel in [parallel_string, ""]:
```

Comment:
very cool, I like the cartesian product. 

nit: perhaps `args_parallel` and `args` might be a better name?